geode-core/src/distributedTest/java/org/apache/geode/internal/cache/wan/asyncqueue/AsyncEventQueuePausedDUnitTest.java

```diff
@@ -0,0 +1,242 @@
+      createRegionAndDispatchingPausedAEQ(props);
+    });
+    client.invoke(() -> {
+      createClientRegion();
+    });
+
+    server1.invoke(() -> {
+      validateAEQDispatchingIsPaused();
+    });
+
+    server2.invoke(() -> {
+      validateAEQDispatchingIsPaused();
+    });
+
+    // Resume dispatching.
+    server1.invoke(() -> {
+      ClusterStartupRule.getCache().getAsyncEventQueue("aeqID").resumeEventDispatching();
+    });
+
+    server2.invoke(() -> {
+      ClusterStartupRule.getCache().getAsyncEventQueue("aeqID").resumeEventDispatching();
+    });
+
+    // Validate dispatching resumed.
+    await().atMost(1, TimeUnit.MINUTES).until(() -> {
+
+      final int count1 = server1.invoke(() -> getEventDispatchedSize());
+      final int count2 = server2.invoke(() -> getEventDispatchedSize());
+      if ((count1 + count2) == 1000) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+
+  }
+
+  @NotNull
+  private static Integer getEventDispatchedSize() {
+    Cache cache = ClusterStartupRule.getCache();
+    AsyncEventQueueImpl aeq = (AsyncEventQueueImpl) cache.getAsyncEventQueue("aeqID");
+    MyAsyncEventListener listener = (MyAsyncEventListener) aeq.getAsyncEventListener();
+    return listener.getEventsMap().size();
+  }
+
+  private static void createClientRegion() {
+    ClientCache cache = ClusterStartupRule.getClientCache();
+    Region region = cache.createClientRegionFactory(ClientRegionShortcut.PROXY)
+        .create("region");
+    for (int i = 0; i < 1000; i++) {
+      region.put(i, i);
+    }
+  }
+
+  private static void createRegionAndDispatchingPausedAEQ(AEQandRegionProperties props) {
+    Cache cache = ClusterStartupRule.getCache();
+    cache.createAsyncEventQueueFactory()
+        .pauseEventDispatching()
+        .setParallel(props.isParallel())
+        .setPersistent(isPersistent(props))
+        .create("aeqID", new MyAsyncEventListener());
+    cache.createRegionFactory(props.getRegionShortcut())
+        .addAsyncEventQueueId("aeqID")
+        .create("region");
+  }
+
+  private static boolean isPersistent(AEQandRegionProperties props) {
+    switch (props.getRegionShortcut()) {
+      case PARTITION:
+      case REPLICATE:
+      case PARTITION_REDUNDANT:
+        return false;
+      default:
+        return true;
+    }
+  }
+
+  private static void validateAEQDispatchingIsPaused() {
+    Cache cache = ClusterStartupRule.getCache();
+    AsyncEventQueueImpl aeq = (AsyncEventQueueImpl) cache.getAsyncEventQueue("aeqID");
+    assertTrue(aeq.isDispatchingPaused());
+    MyAsyncEventListener listener = (MyAsyncEventListener) aeq.getAsyncEventListener();
+    try {
+      await().atMost(10, TimeUnit.SECONDS).until(() -> listener.getEventsMap().size() > 0);
+    } catch (ConditionTimeoutException ex) {
+      // Expected Exception
+    }
+    // Ensure that the the queues are filling up
+    assertTrue(aeq.getSender().getQueues().stream().mapToInt(i -> i.size()).sum() == 1000);
```

Comment:
We could move this before we verify the listener map size; that way we are sure that the AEQ has all the entries, and still the listener is not invoked.

Reply:
It was to avoid the flakiness. Sometimes it does not add up to 1000 immediately because the queuing may take little bit time. So keeping it after the listener validation gives us time buffer.'

Other way I can up an awaitility clause on that assert.